servers/rendering/renderer_rd/storage_rd/particles_storage.cpp

```diff
@@ -452,6 +452,21 @@ void ParticlesStorage::particles_set_transform_align(RID p_particles, RS::Partic
 
 	particles->transform_align = p_transform_align;
 }
+void ParticlesStorage::particles_set_layer_mask(RID p_particles, uint32_t layer_mask) {
+	Particles *particles = particles_owner.get_or_null(p_particles);
+	particles->layer_mask = layer_mask;
+	for (auto &p_collision : particles->collisions) {
+		ParticlesCollision *particle_collision = particles_collision_owner.get_or_null(p_collision);
+		if (particle_collision) {
+			if (!(layer_mask & particle_collision->cull_mask)) {
+				particles->collisions.erase(p_collision);
```

Comment:
If the cull_mask of the particles_collision is updated, it looks like it won't be re-added to the particles' list of collisions. Is that correct? If so, there needs to be an associated check when updating the cull mask for collisions so that they can be added back.

Reply:
That being said, ``particles_add_collision`` may get called when the ``cull_mask`` changes, can you confirm if it does?

Reply:
I added the checks to re-add the collision

Reply:
I added an associative check that re-adds the collision and no particles_add_collision is not called during the `cull_mask` change.